src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1727 @@
+        //
+        unsigned sourceRegCount = 0;
+
+        // To avoid redundant moves, have the argument operand computed in the
+        // register in which the argument is passed to the call.
+
+        for (GenTreeFieldList::Use& use : putArgChild->AsFieldList()->Uses())
+        {
+            GenTree* node = use.GetNode();
+            assert(!node->isContained());
+            // The only multi-reg nodes we should see are OperIsMultiRegOp()
+            assert(!node->IsMultiRegNode());
+
+            // Consume all the registers, setting the appropriate register mask for the ones that
+            // go into registers.
+            // (sourceRegCount < argNode->gtNumRegs)
+            BuildUse(node, argMaskArr[sourceRegCount], 0);
+            sourceRegCount++;
+        }
+        srcCount += sourceRegCount;
+        assert(putArgChild->isContained());
+    }
+    else
+    {
+        assert(putArgChild->TypeGet() == TYP_STRUCT);
+        assert(putArgChild->OperGet() == GT_OBJ);
+
+        // We can use a ldr/str sequence so we need an internal register
+        buildInternalIntRegisterDefForNode(argNode, allRegs(TYP_INT) & ~argMask);
+
+        GenTree* objChild = putArgChild->gtGetOp1();
+        if (objChild->OperGet() == GT_LCL_VAR_ADDR)
+        {
+            // We will generate all of the code for the GT_PUTARG_SPLIT, the GT_OBJ and the GT_LCL_VAR_ADDR
+            // as one contained operation
+            //
+            assert(objChild->isContained());
+        }
+        else
+        {
+            srcCount = BuildIndirUses(putArgChild->AsIndir());
+        }
+        assert(putArgChild->isContained());
+    }
+    buildInternalRegisterUses();
+    assert((argMask != RBM_NONE) && ((int)genCountBits(argMask) == dstCount));
+    for (int i = 0; i < dstCount; i++)
+    {
+        BuildDef(argNode, argMaskArr[i], i);
+    }
+    return srcCount;
+}
+#endif // FEATURE_ARG_SPLIT
+
+//------------------------------------------------------------------------
+// BuildBlockStore: Build the RefPositions for a block store node.
+//
+// Arguments:
+//    blkNode       - The block store node of interest
+//
+// Return Value:
+//    The number of sources consumed by this node.
+//
+int LinearScan::BuildBlockStore(GenTreeBlk* blkNode)
+{
+    GenTree* dstAddr = blkNode->Addr();
+    GenTree* src     = blkNode->Data();
+    unsigned size    = blkNode->Size();
+
+    GenTree* srcAddrOrFill = nullptr;
+
+    regMaskTP dstAddrRegMask = RBM_NONE;
+    regMaskTP srcRegMask     = RBM_NONE;
+    regMaskTP sizeRegMask    = RBM_NONE;
+
+    if (blkNode->OperIsInitBlkOp())
+    {
+        if (src->OperIs(GT_INIT_VAL))
+        {
+            assert(src->isContained());
+            src = src->AsUnOp()->gtGetOp1();
+        }
+
+        srcAddrOrFill = src;
+
+        switch (blkNode->gtBlkOpKind)
+        {
+            case GenTreeBlk::BlkOpKindUnroll:
+                break;
```

Comment:
How is this scenario handled?

Reply:
We reserved the Reg21 for LoongArch64 which not used by lsra, so this can be handled.

Reply:
@kunalspathak 
Thanks very much.

I will update it later.